interceptor/opencensus/opencensus.go

```diff
@@ -89,20 +90,32 @@ func (oci *OCInterceptor) Export(tes agenttracepb.TraceService_ExportServer) err
 		return errTraceExportProtocolViolation
 	}
 
-	var lastNonNilNode *commonpb.Node
+	spansMetricsFn := internal.NewReceivedSpansRecorder("opencensus")
+
+	processReceivedSpans := func(ctx context.Context, ni *commonpb.Node, spans []*tracepb.Span) {
+		// Firstly, we'll add them to the bundler.
+		if len(recv.Spans) > 0 {
+			bundlerPayload := &spansAndNode{node: ni, spans: recv.Spans}
+			traceBundler.Add(bundlerPayload, len(bundlerPayload.spans))
+		}
 
+		// We MUST unconditionally record metrics from this reception.
+		spansMetricsFn(ctx, ni, recv.Spans)
+	}
+
+	var lastNonNilNode *commonpb.Node
 	// Now that we've got the first message with a Node, we can start to receive streamed up spans.
 	for {
 		// If a Node has been sent from downstream, save and use it.
 		if recv.Node != nil {
 			lastNonNilNode = recv.Node
 		}
 
-		// Otherwise add them to the bundler.
-		if len(recv.Spans) > 0 {
-			bundlerPayload := &spansAndNode{node: lastNonNilNode, spans: recv.Spans}
-			traceBundler.Add(bundlerPayload, len(bundlerPayload.spans))
-		}
+		// Note: In normal operations, the context of a streaming RPC usually doesn't
+		// change but we haven't examined all the corner cases for gRPC streaming
+		// hence we'll always pass in the context, in case the underlying implementation
+		// changes a detail. TODO (@odeke-em) investigate if streamer.Context() can change.
```

Comment:
Cannot change and probably it shouldn't.

Reply:
Thanks for the confirmation! I spent sometime yesterday digging through grpc-go to investigate if there any situations where this context can change but didn't seem to find any.

Reply:
@bogdandrutu updated, PTAL.